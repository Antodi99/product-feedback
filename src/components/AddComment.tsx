import { useState } from 'react'
import { createComment } from '../services/comments.service'

type AddCommentProps = {
  id: string
  refreshComments: () => Promise<void>
}

const MAX_INPUT_LENGTH = 250

export function AddComment({ id, refreshComments }: AddCommentProps) {
  const [text, setText] = useState('')

  const handleAddComment = async () => {
    await createComment(text, Number(id))
    refreshComments()
  }

  return (
    <div className='bg-white flex flex-col rounded-lg p-6 h-fit w-full mt-4'>
      <p className='text-dark-blue text-lg font-bold'>Add Comment</p>
      <textarea
        className='bg-light-grey p-2 resize-none w-full outline-none rounded-lg mt-4 h-20 text-dark-blue'
        placeholder='Type your comment here'
        onChange={(e) => {
          setText(e.target.value)
        }}
        maxLength={MAX_INPUT_LENGTH}
        value={text}
      ></textarea>
      <div className='flex justify-between mt-4 items-center text-dark-blue text-sm'>
        <p>{MAX_INPUT_LENGTH - text.length} Characters left</p>
        <div
          onClick={handleAddComment}
          className='rounded-xl p-3 bg-fuchsia-500 w-fit md:ml-3 flex items-center justify-center hover:bg-fuchsia-400 cursor-pointer'
        >
          <p className='text-white font-bold text-xs'>Post Comment</p>
        </div>
      </div>
    </div>
  )
}
