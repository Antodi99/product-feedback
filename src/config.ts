export const BACKEND_API_URL = 'https://go-product-feedback.herokuapp.com'
