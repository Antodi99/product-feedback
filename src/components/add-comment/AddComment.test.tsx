import nock from 'nock'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { render, screen, waitFor } from '@testing-library/react'
import { BACKEND_API_URL } from '../../config'
import { AddComment } from './AddComment'

describe('AddComment', () => {
  it('renders properly', async () => {
    const { container } = render(
      <AddComment id='1' refreshComments={jest.fn()} />
    )
    expect(container).toMatchSnapshot()
  })

  it('handles submit', async () => {
    const scope = nock(BACKEND_API_URL)
      .post('/api/comments/')
      .matchHeader('authorization', /^Bearer .*$/)
      .times(1)
      .reply(200)
    const refreshCommentsMock = jest.fn()
    render(<AddComment id='1' refreshComments={refreshCommentsMock} />)

    const textArea = screen.getByTestId('comment')
    await userEvent.type(textArea, 'Add Comment')
    await userEvent.click(screen.getByText('Post Comment'))
    scope.done()
    await waitFor(() => expect(refreshCommentsMock).toHaveBeenCalledTimes(1))
  })
})
