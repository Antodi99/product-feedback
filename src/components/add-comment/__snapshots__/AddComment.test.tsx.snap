// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddComment renders properly 1`] = `
<div>
  <div
    class="bg-white flex flex-col rounded-lg p-6 h-fit w-full mt-4"
  >
    <p
      class="text-dark-blue text-lg font-bold"
    >
      Add Comment
    </p>
    <textarea
      class="bg-light-grey p-2 resize-none w-full outline-none rounded-lg mt-4 h-auto min-h-[9rem] text-sm text-dark-blue"
      data-testid="comment"
      id="comment"
      name="comment"
      placeholder="Type your comment here"
    />
    <div
      class="flex justify-between mt-4 items-center text-dark-blue text-sm"
    >
      <p>
        250
         Characters left
      </p>
      <div
        class="rounded-xl p-3 bg-fuchsia-500 w-fit md:ml-3 flex items-center justify-center hover:bg-fuchsia-400 cursor-pointer"
      >
        <p
          class="text-white font-bold text-xs"
        >
          Post Comment
        </p>
      </div>
    </div>
  </div>
</div>
`;
